### PR TITLE
Add setup_dev fastlane lane

### PR DESCRIPTION
**Pull request by taquitos**
Add fastlane lane for setup_dev
Add pre-commit hook script that lints files
Link pre-commit during setup_dev lane
Add SwiftLint to new build phase step in project so people can see issues before commit.
Update documentation

### CODE REVIEW

**Comment by taquitos** [comment]
Right now the pre-commit will auto-fix, but it won't stop the commit.

**Comment by taquitos** [comment]
Example of warning in Xcode 
Screen Shot 2021-06-11 at 4 27 23 PM

File: Purchases.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1586,6 +1587,26 @@
...
+			);
+			inputPaths = (
+			);
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "scripts/swiftlint.sh\n";
```
Comment (aboedo) [inline Purchases.xcodeproj/project.pbxproj:1606]: 👍

**Review by aboedo — APPROVED** [review]
👏👏👏👏

**Comment by taquitos** [comment]
Updated to autofix and stop the commit with a friendly message
Screen Shot 2021-06-15 at 9 33 09 AM

**Review by beylmk — APPROVED** [review]
this is amazing, thank you!

**Pull request merged**